client/src/app/project-form/project-form.component.ts

```diff
@@ -85,9 +86,14 @@ export class ProjectFormComponent implements OnInit {
       .subscribe((account) => {
         this.userIsWrangler = account.isWrangler();
         this.setUpProjectForm(this.userIsWrangler, layout, this.createMode, pathVariables);
+        this.userAccount = account;
```

Comment:
I think you could just set an `initialised` property to `true` here instead of the function right? Seems a bit easier to reason about. Then you know that you just wait for the subscription to complete for it to be initialized without having to think about what `isInitialised` is checking.

Reply:
ok, that could be cleaner, I agree.
TBH I was thinking about the same thing. ;-)

Reply:
Well, I just realised that it is not that simple, because the `schema` variable is also created asynchronously. So, It would need more change.

Reply:
Okay, well we should do that change as part of [#225](https://app.zenhub.com/workspaces/dcp-ingest-product-development-5f71ca62a3cb47326bdc1b5c/issues/ebi-ait/dcp-ingest-central/225). Ideally, we should be using Rx pipes to chain observables rather than having intermediate variables and hard to track cross-dependencies